provider/godaddy/client.go

```diff
@@ -77,6 +77,29 @@ type Client struct {
 	Timeout time.Duration
 }
 
+// GDErrorField describe the error reason
+type GDErrorField struct {
+	Code        *string `json:"code,omitempty"`
+	Message     *string `json:"message,omitempty"`
+	Path        *string `json:"path,omitempty"`
+	PathRelated *string `json:"pathRelated,omitempty"`
+}
+
+// GDErrorResponse is the body response when an API call fails
+type GDErrorResponse struct {
+	Code    string         `json:"code"`
+	Fields  []GDErrorField `json:"fields,omitempty"`
+	Message string         `json:"message,omitempty"`
+}
+
+func (r GDErrorResponse) String() string {
+	if b, err := json.Marshal(r); err == nil {
```

Comment:
why not returning the err? IMHO it should be still handled correctly, debugging and figuring out what is wrong might be time consuming.

Reply:
Answer is same as similar toString(), except no indentation in the json. Also the  String() function is a standard to represent a structure in string form and it doesn't have 2 valued return.